### PR TITLE
feat(input): [NoTicket] Extend input

### PR DESCRIPTION
**Pull request by busypeoples**
### What this PR does

* Add a new `Label` property to the Input component

### Checklist:

- [ ] I reviewed my own code
- [ ] The changes align with the designs I received  
  Or give a reason why this does not apply:
- [ ] I have attached screenshot(s), video(s) or gif(s) showing that the solution is working as expected  
  Or give a reason why this does not apply:
- [ ] I have updated the task(s) status on Linear
- [ ] All new media is optimized (images, gifs, videos)

### Browser support

My code works in the following browsers:

- [ ] Firefox
- [ ] Chrome
- [ ] Safari
- [ ] Edge

### CODE REVIEW

**Review by talo242 — CHANGES_REQUESTED** [review]

File: src/lib/components/input/index.tsx [hunk 1]
```diff
@@ -7,12 +7,21 @@ import styles from './style.module.scss';
...
 } & Omit<JSX.IntrinsicElements['input'], 'enterKeyHint'> &
   Partial<Pick<JSX.IntrinsicElements['input'], 'enterKeyHint'>>;
 
 export default React.forwardRef(
   (
-    { className, placeholder, prefix, error, disabled, ...props }: InputProps,
+    {
+      className,
+      placeholder,
+      placeholderAsLabel = true,
```
[the remark this answers is not in the document]
Comment (busypeoples) [reply src/lib/components/input/index.tsx:19]: Agreed regarding `label` being needed for accessibility.

[the remark this answers is not in the document]
Comment (busypeoples) [reply src/lib/components/input/index.tsx:19]: @talo242 this makes a lot of sense. What is the best way forward here? Should we add the `label` property and make it backwards compatible ensuring that the current way this component works doesn't change if no `label` is provided?

[the remark this answers is not in the document]
Comment (kimkwanka) [reply src/lib/components/input/index.tsx:19]: Another related problem is that the actual `placeholder` attribute on the `input` is set to `" "` which makes it "untargetable" via React Testing Library for example. This however, does not actually affect `a11y` as screen readers [don't seem to read the placeholder](https://github.com/microsoft/fluentui/issues/16583#issuecomment-767164317).

[the remark this answers is not in the document]
Comment (busypeoples) [reply src/lib/components/input/index.tsx]: With the current implementation if you provide a `label`, the input will use the actual `placeholder`. In the future we should make the `label` required.

**Review by talo242 — APPROVED** [review]
Looking good! ✨

**Review by ruizhen88 — APPROVED** [review]

**Review by kimkwanka — APPROVED** [review]

**Pull request merged**